core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -517,6 +518,32 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with Timeou
     assertDataStructuresEmpty()
   }
 
+  test("unserializable partition") {
```

Comment:
Also, I defer to @squito, but I think this might be a better fit for SchedulerIntegrationSuite, since it's not testing any of the DAGScheduler's functionality.  Alternately, you could it in CoarseGrainedSchedulerBackendSuite and mock the TaskSetManager (to ensure abort() gets called).